auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -503,15 +504,24 @@ public String rightCheckToString(AuthorizationProto.RightCheck rightCheck) {
         }
     }
 
-    @Cacheable(cacheNames = "umsAuthorizationEntitlementRegisteredCache", key = "{ #actorCrn, #accountId }")
-    public boolean isAuthorizationEntitlementRegistered(String actorCrn, String accountId) {
-        if (StringUtils.equals(accountId, InternalCrnBuilder.INTERNAL_ACCOUNT)) {
-            return false;
+    @Cacheable(cacheNames = "umsEntitlementRegisteredCache", key = "{ #actorCrn, #accountId, #entitlement }")
+    public boolean isEntitled(String actorCrn, String accountId, Entitlement entitlement) {
+        boolean entitled;
+        if (Entitlement.CB_AUTHZ_POWER_USERS.equals(entitlement) && StringUtils.equals(accountId, InternalCrnBuilder.INTERNAL_ACCOUNT)) {
+            entitled = false;
```

Comment:
maybe it is better to fail here if account id is internal (which should not ever happen), so i would not check for that, anyway CB will fail somewhere else also if internal account id reaches service layer

Reply:
nvm i put that kind of if statement because of an issue